src/libs/Network.js

```diff
@@ -161,6 +186,16 @@ function processNetworkRequestQueue() {
             .catch(error => onError(queuedRequest, error));
     });
 
+    // We should clear the NETWORK_REQUEST_QUEUE when we have loaded the perssited requests & Client is leader
```

Comment:
Sorry, I still don't get why we are doing this or **_why_** we care that the "client is the leader" from reading this comment